src/vchordrq/algorithm/build.rs

```diff
@@ -203,14 +203,34 @@ impl Structure {
     ) -> Vec<Self> {
         use std::collections::BTreeMap;
         let VchordrqExternalBuildOptions { table } = external_build;
-        let query = format!("SELECT id, parent, vector FROM {table};");
+        let dump_query = format!("SELECT id, parent, vector FROM {table};");
+        let table_name = table.split('.').last().unwrap().to_string();
+        let type_check_query = format!(
```

Comment:
We'd better get schema name of extension vector here by querying `pg_extension` and `pg_namespace`. So that we could get oid of `vector` by `regtypein('SCHEMA_NAME.vector')` (and `halfvec`).

If we want to support `halfvec`, we could then query

```
SELECT atttypid FROM pg_attribute WHERE attrelid = '{table}'::regclass AND attname = 'vector';
```

to know what type it exactly is.

If we don't want to support `halfvec`, we could just query `SELECT id, parent, vector::{vector_schema}.vector FROM {table};` for centroids.

Reply:
OK, I prefer to not support halfvec and force cast it.